abixen-platform-modules/src/main/java/com/abixen/platform/module/chart/service/impl/DatabaseFactoryImpl.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2010-present Abixen Systems. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.abixen.platform.module.chart.service.impl;
+
+import com.abixen.platform.module.chart.exception.DatabaseConnectionException;
+import com.abixen.platform.module.chart.model.enumtype.DatabaseType;
+import com.abixen.platform.module.chart.service.DatabaseFactory;
+import com.abixen.platform.module.chart.service.DatabaseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DatabaseFactoryImpl implements DatabaseFactory {
+
+    @Autowired
+    private DatabasePostgresServiceImpl databasePostgresService;
```

Comment:
Change to:

@Autowired
@Qualifier("databasePostgresService")
 private DatabaseService databasePostgresService;

@Autowired
@Qualifier("databaseH2Service")
private DatabaseService databaseH2Service;